NEWS.md

```diff
@@ -1,3 +1,8 @@
+## 5.0.0 IN PROGRESS
+ * [FOLSPRINGB-58](https://issues.folio.org/browse/FOLSPRINGB-58) Async FolioExecutionContext helper
+
+Breaking change: Replace calls to `FolioExecutionScopeExecutionContextManager.beginFolioExecutionContext` and  `FolioExecutionScopeExecutionContextManager.endFolioExecutionContext` with `FolioExecutionContextSetter` usage.
```

Comment:
Could you please consider necessity of the term "Breaking change" in this file. AFAIK, the term "Breaking change" means, that the contract of the previous version is broken. The contract could be broken, if the code doesn't work with the new version of the dependency. But it is not the case for this PR. And it cannot be the case, because the `beginFolioExecutionContext` cannot be made `private`, because FolioExecutionContextSetter cannot be used in some special cases.

Reply:
Without the breaking change developers will continue to use the error-prone `beginFolioExecutionContext` and `endFolioExecutionContext` in existing and new code.
We need a [nudge](https://en.wikipedia.org/wiki/Nudge_theory) to change this.

The breaking change ensures that
* all `beginFolioExecutionContext` are handled, otherwise the compiler gives a visibility error
* a missing folioExecutionContextSetter.close() gives an [autoclosable warning in static code analyzers like Sonar](https://sonarcloud.io/organizations/folio-org/rules?q=autoclosable&open=java%3AS2095) in cases where no try-with-resources is used.

I've added an example without try-with-resources to README.md and FolioExecutionContextSetterTest.java.